### ISSUE TITLE
Extend keyword false positives for yaml/helm template syntax

### ISSUE THREAD

**Issue by mikkelblanne** [Yelp/detect-secrets#360]
We have a lot of configuration in yaml files containing helm chart templates, with secrets being templated and inserted afterwards. The basic template construct `{{var}}` is [already covered](https://github.com/Yelp/detect-secrets/blob/f1c55bbf387e8b07813a2872779a2246bfd1d90f/detect_secrets/plugins/keyword.py#L140), but the templates allow spaces within the brace expression, and the way keyword parsing [is done](https://github.com/Yelp/detect-secrets/blob/f1c55bbf387e8b07813a2872779a2246bfd1d90f/detect_secrets/plugins/keyword.py#L164), results in the extracted secret being split on the first space, and include the optional quote.

The result is that given an input line like this:
`password: "{{ var }}"`
the extracted secret becomes `"{{`.

Hence it would solve our many false positivies, if these prefixes were added to [FALSE_POSITIVES](https://github.com/Yelp/detect-secrets/blob/f1c55bbf387e8b07813a2872779a2246bfd1d90f/detect_secrets/plugins/keyword.py#L50):
`    '"{{',`
`    '\'{{',`

I don't think that would filter out any real secrets. Another possibility is to change the regex to actually allow spaces within the secret value, but that looks like a major change to the keyword detector.

**Comment by domanchi** [comment]
@mikkelblanne, can you write a test case to clearly indicate a successful case where it should pass, and one where it should fail?

**Comment by mikkelblanne** [comment]
@domanchi, I tried looking at keyword_test, but not having ever really developed in python, I gave up on actually running the tests. But simply put (just talking about adding the two prefixes), I'd like these to result in negatives:

`"apikey: '{{ helm_secret_key }}'",`
`'apikey: "{{ helm_secret_key }}"',`
`'apikey: {{ helm_secret_key }}',`

I guess they belong in FOLLOWED_BY_COLON_RE.

The last one should already pass, because the secret becomes `{{`, which is already in FALSE_POSITIVES.

Is that sufficient? I'm not talking about adding any new negatives, so no new tests for that.

**Comment by domanchi** [comment]
Yeah, I agree the `keyword_test.py` file is really hard to grok through. Your proposed test cases make your ask clearer. Thanks!

**Comment by mikkelblanne** [comment]
Thanks for taking a look!

Well, it's not just the grokability of the tests themselves, I think I figured it out. But I have zero experience with virtualenv, and when I tried to run pytest directly, it complained about needing pyahocorasick, and I'm on Windows, so that install fails, complaining that it requires Visual C++ 14.0. Which I have... And someone created [an issue](https://github.com/WojciechMula/pyahocorasick/issues/129) about that half a year ago, and so far has not received any response :-/  I guess you really need linux for doing anything serious in python.

**Comment by KevinHock** [comment]
Sorry about the headaches you've mentioned

Would it help if we didn't report anything of length 4 or greater via the keyword detector? I think that would capture this.
https://github.com/Yelp/detect-secrets/issues/324#issuecomment-679323365

**Comment by mikkelblanne** [comment]
Assuming that you meant "anything of length 4 **or less**, you're right :)

Due to the space parsing rule, that would work as well. And I guess secrets with less than 5 characters are rarely used for anything important, except perhaps pin codes.

**Comment by omobolajiadeyan** [comment]
I would like to work on this. I will add regression coverage for quoted Helm template values and keep the implementation scoped to the keyword detector false-positive handling.

**Comment by omobolajiadeyan** [comment]
I tested this against current `main` (`5e14193`) before editing. The single-quoted, double-quoted, and unquoted Helm examples all return no findings from `KeywordDetector`, and `python -m pytest tests/plugins/keyword_test.py -q` passes all 351 tests. It appears the newer keyword regex/templated-secret handling resolved this case, so I am stepping back rather than submitting a no-op change.
